java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java

```diff
@@ -212,32 +214,66 @@ private static List<SymbolInformation> search(String projectName, ArrayList<Stri
         }
 
 		IJavaSearchScope scope;
+        var workspaceDirectoryLocation = JavaLanguageServerPlugin.getPreferencesManager().getPreferences().getRootPaths();
+        if (workspaceDirectoryLocation == null || workspaceDirectoryLocation.size() == 0) {
+            logInfo("unable to find workspace directory location");
+            return new ArrayList<>();
+        }
+        
         if (includedPaths != null && includedPaths.size() > 0) {
             ArrayList<IJavaElement> includedFragments = new ArrayList<IJavaElement>();
             for (IJavaProject proj : targetProjects) {
-                for (IPackageFragment fragment : proj.getPackageFragments()) {
-                    IPath fragmentPath = fragment.getPath();
-                    // if there's no file extension, it's a path to java package from source
-                    // else it is a path pointing to jar, ear, etc. we ignore deps for now
-                    if (fragmentPath.getFileExtension() == null) {
+                for (String includedPath : includedPaths) {
+                    IPath includedIPath = Path.fromOSString(includedPath);
+                    if (includedIPath.isAbsolute()) {
+                        includedIPath = includedIPath.makeRelativeTo(workspaceDirectoryLocation.iterator().next());
+                        // we need to remove the /src/java from the path
+                        if (includedIPath.segment(0).equals("src")) {
+                            includedIPath = includedIPath.removeFirstSegments(1);
+                        }
+                        if (includedIPath.segment(0).equals("main")) {
+                            includedIPath = includedIPath.removeFirstSegments(1);
+                        }
+                        if (includedIPath.segment(0).equals("java")) {
+                            includedIPath = includedIPath.removeFirstSegments(1);
+                        }
+                        var element = proj.findElement(includedIPath);
+                        if (element == null) {
+                            element = proj.findElement(includedIPath.removeLastSegments(1));
+                            continue;
+                        }
+                        if (element instanceof ICompilationUnit) {
+                            var x = element.getAncestor(IJavaElement.PACKAGE_FRAGMENT);
+                            if (x != null) {
+                                includedFragments.add(x);
+                            }
+                        } else {
+                            includedFragments.add(element);
+                        }
+                        continue;
+                    }
+                    for (IPackageFragment fragment : proj.getPackageFragments()) {
+                        IPath fragmentPath = fragment.getPath();
+                        // if there's no file extension, it's a path to java package from source
```

Comment:
Can't we send `*.java` files also? I was taking a look the other day but couldn't find the way